### PR TITLE
Fix late nem farmining issues

### PR DESCRIPTION
**Pull request by jlward**

### CODE REVIEW

File: nemesis/arenaPrep.py [hunk 1]
```diff
@@ -1121,8 +1118,8 @@ def kilika_final_shop():
     menu.auto_sort_equipment()
 
     gilNeeded = 3500000 - memory.main.get_gil_value()
-    weaponBuys = int(gilNeeded / 26150)
-    weaponBuys += 1  # for safety
+    # Get minimum needed, plus one for safety. Max 99 total.
+    weaponBuys = min(int(gilNeeded / 26150), 98) + 1
```
Comment (jlward) [inline nemesis/arenaPrep.py:1122]: This was a fix pitched by @coderwilson and it seemed to work on my machine at least once.

File: nemesis/arena_battles.py [hunk 2]
```diff
@@ -822,7 +822,7 @@ def item_dump():
 
 def quick_reset_logic():
     reset.reset_to_main_menu()
-    memory.ain.wait_frames(90)
+    memory.main.wait_frames(90)
```
Comment (jlward) [inline nemesis/arena_battles.py:825]: This is the only thing that hasn't been tested already. However, this change feels fairly straightforward.

File: main.py [hunk 3]
```diff
@@ -672,7 +672,10 @@ def perform_TAS():
 
                 if game.step == 13:
                     nemesis.arenaPrep.djose_farm(cap_num=10)
-                    # NOTE There does not appear to be a step 14 for Nem Farming.
+                    game.step = 14
+
+                if game.step == 14:
+                    nemesis.arenaPrep.t_plains(cap_num=10)
```
Comment (jlward) [inline main.py:678]: This has been tested and works.

File: nemesis/arena_battles.py (hunk 2 above)
Comment (jlward) [reply nemesis/arena_battles.py:828]: Barring a premature death or profound luck, this should be tested in an hour or two.

Comment (jlward) [reply nemesis/arena_battles.py:828]: This just got tested, it's working.

File: nemesis/arena_battles.py [hunk 4]
```diff
@@ -461,7 +462,7 @@ def battles_1():
     restock_downs()
     nemesis.arenaSelect.arena_menu_select(4)
     memory.main.full_party_format("kilikawoods1")
-    nemesis.menu.tidusSlayer(odPos=0)
+    menu.tidus_slayer(odPos=0)
```
Comment (jlward) [inline nemesis/arena_battles.py:465]: These change have not been tested. It'll take me two hours to get here again. I really ought to figure out how to get a save at the start of nem arena. So I can just jump right here.

File: nemesis/arenaPrep.py [hunk 5]
```diff
@@ -3044,13 +3041,13 @@ def gagazet(cap_num: int = 10):
                 checkpoint = 12
 
             # NEA decisions
-            if neArmor == True and checkpoint in [9, 17]:
+            if neArmor == True and checkpoint in [7,18]:
```
Comment (jlward) [inline nemesis/arenaPrep.py:3044]: These two guys have been tested.

File: nemesis/arenaPrep.py (hunk 1 above)
Comment (coderwilson) [reply nemesis/arenaPrep.py:1125]: I'm editing this further for an "out of gil" scenario, but this should be fine to merge.

**Pull request merged**